Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -555,3 +588,83 @@ void ABenchmarkGymGameModeBase::ReportAuthoritativePlayers_Implementation(const
 		}
 	}
 }
+
+void ABenchmarkGymGameModeBase::TickActorMigration(float DeltaSeconds)
+{
+	if (bHasActorMigrationCheckFailed)
+	{
+		return;
+	}
+
+	const UNFRConstants* Constants = UNFRConstants::Get(GetWorld());
+	check(Constants);
+
+	// This test respects the initial delay timer in both native and GDK
+	if (Constants->ActorMigrationCheckDelay.HasTimerGoneOff())
+	{
+		// Count how many actors hand over authority in 1 tick
+		int Delta = FMath::Abs(UXAuthActorCount - PreviousTickMigration);
+		ToBeRemovedMigrationDeltas.Enqueue(ToBeRemovedDelta(Delta, DeltaSeconds));
+		bool bChanged = false;
+		if (MigrationCountSeconds > MigrationWindowSeconds)
+		{
+			ToBeRemovedDelta RemoveValue;
+			if (ToBeRemovedMigrationDeltas.Peek(RemoveValue))
+			{
+				ToBeRemovedMigrationDeltas.Pop();
+				MigrationOfCurrentWorker -= RemoveValue.Key;
+				MigrationSeconds -= RemoveValue.Value;
+				bChanged = (Delta != RemoveValue.Key);
+			}
+		}
+		MigrationOfCurrentWorker += Delta;
+		MigrationSeconds += DeltaSeconds;
+		PreviousTickMigration = UXAuthActorCount;
+
+		if (MigrationCountSeconds > MigrationWindowSeconds)
+		{
+			if (bChanged)
+			{
+				// Only report MigratedActorsOfCurrentWorker to the worker which has authority and the Migration changed
+				float AverageMigrationOfCurrentWorker = MigrationOfCurrentWorker / MigrationSeconds;
+				ReportMigration(FPlatformProcess::ComputerName(), AverageMigrationOfCurrentWorker);
+			}
+
+			if (HasAuthority() && ActorMigrationCheckTimer.HasTimerGoneOff())
+			{
+				if (MapWorkerActorMigration.Num() > 1)
+				{
+					float TotalMigrations = 0;
+					for (const auto& KeyValue : MapWorkerActorMigration)
+					{
+						TotalMigrations += KeyValue.Value;
+					}
+					float AverageActorMigration = TotalMigrations / MapWorkerActorMigration.Num();
+					if (AverageActorMigration < MinActorMigrationPerSecond)
+					{
+						bHasActorMigrationCheckFailed = true;
+						NFR_LOG(LogBenchmarkGymGameModeBase, Error, TEXT("%s: Actor migration check failed. TotalMigrations=%.3f AverageActorMigration=%.3f MinActorMigrationPerSecond=%.3f MigrationExactlyWindowSeconds=%.3f"),
+							*NFRFailureString, TotalMigrations, AverageActorMigration, MinActorMigrationPerSecond, MigrationSeconds);
+					}
+					else
+					{
+						// Reset timer for next check after 10s
+						ActorMigrationCheckTimer.SetTimer(10);
+						UE_LOG(LogBenchmarkGymGameModeBase, Log, TEXT("Actor migration check TotalMigrations=%.3f AverageActorMigration=%.3f MinActorMigrationPerSecond=%.3f MigrationExactlyWindowSeconds=%.3f"),
+							TotalMigrations, AverageActorMigration, MinActorMigrationPerSecond, MigrationSeconds);
+					}
+				}
+			}
+		}
+		MigrationCountSeconds += DeltaSeconds;
+	}
+}
+
+void ABenchmarkGymGameModeBase::ReportMigration_Implementation(const FString& WorkerID, const float AverageMigration)
+{
+	if (HasAuthority())
+	{
+		float& Value = MapWorkerActorMigration.FindOrAdd(WorkerID);
```

Comment:
I didn't ask this in the PR for connected clients, but why use FindOrAdd?
Why not use .Emplace? operator [] ..